### DISCUSSION TITLE
TShock in Docker/Linux crashes at startup since 4.5.5

### DISCUSSION THREAD

**Discussion by ghost** [Pryaxis/TShock#2524] [Q&A] [answered]
Hello,

First off, thank you for your all work on TShock, I really appreciate and enjoy the enhancements it brings to Terraria over the vanilla server.

Ever since version 4.5.5 my TShock server container has been unusable. Version 4.5.4 and prior are stable but starting in version 4.5.6 I get "ERROR: System.NullReferenceException: Object reference not set to an instance of an object".

A bit of background first. I normally run TShock as a container on a Linux VM and this has been working perfectly for a while now. If I run TShock natively under Windows using the same TShock zip and world file all is well.

The Docker image is amd64/debian:11.1 based with with mono-complete 6.12.122 installed from the mono apt repository (standard process for installing mono, _not_ the mono version from a Debian/3rd party repo). Again, this image has been working perfectly in 4.5.5 and older so I don't think it's missing anything or using incompatible versions (unless something changed as of 4.5.6).

I've also tried copying what other Docker TShock projects do that I found online which use a mono image as the base as well as rolling back to Debian 10 and trying alternative mono versions but I get the same results in all cases.

I get an identical stack trace regardless if I use the world I've been playing since Journey's End came out or I generate a clean world in the current game version.

My question is, did something change in 4.5.6 that requires changes for Linux/Mono servers?

Thanks

Log Files (this is a fresh run, new world file using the latest 4.5.11 version)
[crash_132827830833728110.zip](https://github.com/Pryaxis/TShock/files/7629141/crash_132827830833728110.zip)
[2021-11-30_21-53-43.log](https://github.com/Pryaxis/TShock/files/7629142/2021-11-30_21-53-43.log)

I can make my Dockerfile and entry script available if needed but that won't entirely make sense outside my environment (I run a private Gitlab instance and use Gitlab's CI/CD tooling to provide some of the environmental details but I might be able to substitute equivalent parameters for sake of clarity).

**Comment by hakusaro** [comment, marked as answer]
The only possible thing I can think of is that we're keeping console input active. Can you pass `-disable-commands` as a startup argument to TShock and try booting again?

**Reply by ghost** [reply]
Thank you. That appears to have been the issue. I don't have a whole lot of time to test right now but a quick smoke test shows that the containerized TShock is running on 4.5.11 and I was able to join the server in-game. I will do some more thorough testing later when I get a chance but I think this solved it.

Verified with TShock 4.5.11 on Mono 6.12.0.122 in a custom Debian 11.1 Docker container.

For reference my command to launch TShock now looks like (the variables are provided by the environment's pipeline, adjust to suit your needs, these are all just file paths anyway):

mono --server --gc=sgen -O=all TerrariaServer.exe --stats-optout **_-disable-commands_** -configpath  "$CONFIG_PATH" -crashdir "${CONFIG_PATH}/logs/crash" -worldpath "$WORLD_PATH" -world "${WORLD_PATH}/${WORLD_NAME}.wld"

**Reply by hakusaro** [reply]
@thesuperfuzz that is awesome to know! Thank you for testing!
